File: 프로그래머스/lv2/42842. 카펫/카펫.py
def solution(brown, yellow):
    #가로 >= 세로
    answer = []
    num = int(yellow ** 0.5)
    w = 1
    h = 1
    for n in range(1,num+1):
        print(n)
        if ((n*2) + ((yellow // n) * 2) + 4) == brown:
            w = yellow // n
            h = n
        
    answer = [w+2, h+2]
    return answer

File: 프로그래머스/lv2/42842. 카펫/카펫.js
function solution(brown, yellow) {
    let answer = [0, 0]
    
    for (let w = 1; w <= yellow; w++){
        let h = Math.floor(yellow/w)
        
        if(w >= h) {
            let size = (w*2) + (h*2) + 4;
            
            if (brown == size) {
                answer = [w + 2, h + 2]
                break
            }
        }
    }
    
    
    return answer
}